lib/src/edhoc.rs

```diff
@@ -853,20 +879,10 @@ fn encode_plaintext_2(
 ) -> Result<BufferPlaintext2, EDHOCError> {
     let mut plaintext_2: BufferPlaintext2 = BufferPlaintext2::new();
     let c_r = c_r.as_slice();
-    plaintext_2.content[..c_r.len()].copy_from_slice(c_r);
 
-    let offset_cred = match id_cred_r {
-        IdCred::CompactKid(kid) => {
-            plaintext_2.content[c_r.len()] = *kid;
-            c_r.len() + 1
-        }
-        IdCred::FullCredential(cred) => {
-            plaintext_2.content[c_r.len()] = CBOR_BYTE_STRING;
-            plaintext_2.content[c_r.len() + 1] = cred.len() as u8;
-            plaintext_2.content[c_r.len() + 2..][..cred.len()].copy_from_slice(cred);
-            c_r.len() + 2 + cred.len()
-        }
-    };
+    plaintext_2.extend_from_slice(c_r).unwrap();
```

Comment:
I know that the `copy_from_slice` could already lead to panicking behavior, but we could improve it and avoid the explicit `unwrap` here by:

```diff
diff --git a/lib/src/edhoc.rs b/lib/src/edhoc.rs
index d29e01b..b793258 100644
--- a/lib/src/edhoc.rs
+++ b/lib/src/edhoc.rs
@@ -880,7 +880,7 @@ fn encode_plaintext_2(
     let mut plaintext_2: BufferPlaintext2 = BufferPlaintext2::new();
     let c_r = c_r.as_slice();
 
-    plaintext_2.extend_from_slice(c_r).unwrap();
+    plaintext_2.extend_from_slice(c_r).or(Err(EDHOCError::EncodingError))?;
     id_cred_r.write(&mut plaintext_2)?;
     let offset_cred = plaintext_2.len;
 
diff --git a/shared/src/lib.rs b/shared/src/lib.rs
index 9e29870..5bee193 100644
--- a/shared/src/lib.rs
+++ b/shared/src/lib.rs
@@ -193,6 +193,7 @@ pub enum EDHOCError {
     UnsupportedMethod,
     UnsupportedCipherSuite,
     ParsingError,
+    EncodingError,
     CredentialTooLongError,
     EadLabelTooLongError,
     EadTooLongError,
@@ -229,6 +230,7 @@ impl EDHOCError {
             UnsupportedMethod => ErrCode::UNSPECIFIED,
             UnsupportedCipherSuite => ErrCode::WRONG_SELECTED_CIPHER_SUITE,
             ParsingError => ErrCode::UNSPECIFIED,
+            EncodingError => ErrCode::UNSPECIFIED,
             CredentialTooLongError => ErrCode::UNSPECIFIED,
             EadLabelTooLongError => ErrCode::UNSPECIFIED,
             EadTooLongError => ErrCode::UNSPECIFIED,
```

Reply:
LGTM, applied.